libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.NetFx/BotActivitiesHandler.cs

```diff
@@ -0,0 +1,99 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+using Microsoft.Bot.Builder.Adapters;
+using Microsoft.Bot.Schema;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Formatting;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Bot.Builder.Integration.AspNet
+{
+    internal sealed class BotActivitiesHandler : HttpMessageHandler
+    {
+        private static readonly MediaTypeFormatter[] BotActivityMediaTypeFormatters = new [] {
+            new JsonMediaTypeFormatter
+            {
+                SerializerSettings =
+                {
+                    ContractResolver = new CamelCasePropertyNamesContractResolver(),
+                    Formatting = Newtonsoft.Json.Formatting.Indented,
+                    NullValueHandling = NullValueHandling.Ignore,
+                }
+            }
+        };
+
+        private readonly BotFrameworkAdapter _botFrameworkAdapter;
+
+        public BotActivitiesHandler(BotFrameworkAdapter botFrameworkAdapter)
+        {
+            _botFrameworkAdapter = botFrameworkAdapter;
+        }
+
+        protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            if (request.Method != HttpMethod.Post)
+            {
+                return request.CreateResponse(HttpStatusCode.MethodNotAllowed);
+            }
+
+            var requestContent = request.Content;
+            var requestContentHeaders = requestContent.Headers;
+
+            if (requestContentHeaders.ContentLength == 0)
+            {
+                return request.CreateErrorResponse(HttpStatusCode.BadRequest, "Request body should not be empty.");
+            }
+
+            if (!BotActivityMediaTypeFormatters[0].SupportedMediaTypes.Contains(requestContentHeaders.ContentType))
+            {
+                return request.CreateErrorResponse(HttpStatusCode.NotAcceptable, $"Expecting Content-Type of \"{BotActivityMediaTypeFormatters[0].SupportedMediaTypes[0].MediaType}\".");
+            }
+
+            var activity = await requestContent.ReadAsAsync<Activity>(BotActivitiesHandler.BotActivityMediaTypeFormatters, cancellationToken);
+
+            try
+            {
+                await _botFrameworkAdapter.ProcessActivity(
+                    request.Headers.Authorization?.Parameter,
+                    activity,
+                    botContext =>
+                    {
+                        cancellationToken.ThrowIfCancellationRequested();
+
+                        IBot bot;
+
+                        try
+                        {
+                            bot = (IBot)request.GetDependencyScope().GetService(typeof(IBot));
```

Comment:
This assumes that the context has IBot attached to it right (or DI into container)? While I understand the reason for doing this all ourselves but are we going too far? Are we approaching the territory of over engineering? I understand that someone can just use go ahead and not use this integration and use the adapter directly but that is just too many decisions involved before they even start bot development. This integration probably makes more sense as a sample so that is can guide people to do the right thing.